src/pages/DoctorDetail.tsx

```diff
@@ -0,0 +1,68 @@
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { specialityImages } from "../assets/assets";
+import { MdVerified } from "react-icons/md";
+import { IoIosInformationCircleOutline } from "react-icons/io";
+import { fetchDoctors } from "../components/services/patientService";
+
+export function DoctorDetail() {
+  const { docId } = useParams();
+  const [docInfo, setDocInfo] = useState(null);
+
+  useEffect(() => {
+    const loadDoctor = async () => {
+      const allDoctors = await fetchDoctors();
+      const doctor = allDoctors.find((doc) => doc.id === docId);
+      setDocInfo(doctor || null);
+    };
+    loadDoctor();
+  }, [docId]);
```

Comment:
A czemu nie zrobisz tego również useSWR? Możesz zrobić komponent pośredniczący, który pobiera docId, wywołuje useSWR, zamiast komponentu ze szczegółami renderuje loading, a jak wszystko sie skonczy, to amiast loadinga wyswietla dodatkowy komponent ktoremu przekazesz dane doktora i on wyswietli wszystko :P